api/models.py

```diff
@@ -229,6 +264,74 @@ def to_dict(self, for_user=None):
 
         return d
 
+class UserSkill(db.Model):
+    __tablename__ = 'user_skills'
+
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+
+    user_id = db.Column(db.Integer, db.ForeignKey(User.id), nullable=False)
+    skill = db.Column(db.String(21))
+
+    added_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+
+class UserAchievement(db.Model, GeneratedKeyMixin):
+    __tablename__ = 'user_achievements'
+
+    REQUIRED_FIELDS = ['achievement']
+
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    key = db.Column(db.String(64), nullable=False)
```

Comment:
Why do we need id AND key?

I see you use the key to do a lot, but could the ID be used instead?

Reply:
The reason I did this is:

Numeric auto-increment IDs are easy to work with locally in the database - easy to make queries when debugging, etc, faster queries JOINing on foreign keys... and generally more convenient. I think it is just generally considered best practice to use auto-increment IDs in SQL databases.

HOWEVER

I don't want to expose the numeric IDs externally, for multiple reasons.

Now, the `key` we currently have is not really great either - it is a very short string, and can repeat between different tables (an auction and a listing can have the same key). But the good part is that we do have the mechanism in place, and all we need to do is replace the function that generates the key with something else, which I think we should do soon.

Ideally we would have a key that contains enough entropy to be unique across all instances of PM, so it could be used to uniquely identify something in the whole network.

Reply:
Yes, we'll think into that very soon.